src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -747,6 +787,18 @@ private static unsafe nuint GetIndexOfFirstNonAsciiChar_Sse2(char* pBuffer, nuin
                             goto FoundNonAsciiDataInFirstOrSecondVector;
                         }
                     }
+                    else if (AdvSimd.Arm64.IsSupported)
+                    {
+                        currentMask = Unicode.Utf16Utility.GetNonAsciiBytes(AdvSimd.AddSaturate(combinedVector, asciiMaskForAddSaturate).AsByte(), bitmask);
```

Comment:
Minor nit: The `goto FoundNonAsciiDataInFirstOrSecondVector;` statement is replicated across many branches, which could make the logic harder to modify longer-term. Consider factoring SIMD logic into a predicate method, e.g. `bool ContainNonAsciiDataInFirstOrSecondVector()` and then consume like so:

```csharp
if (ContainNonAsciiDataInFirstOrSecondVector(..args..))
{
    goto FoundNonAsciiDataInFirstOrSecondVector;
}
```